src/Entity/Company.php

```diff
@@ -55,6 +57,14 @@ public function __construct()
         $this->jobOffers = new ArrayCollection();
     }
 
+    public static function createFromRegistration(CompanyRegistration $registration): self
```

Comment:
This method should be in the DTO class. It is the DTO responsibility to create the real object because If you have many DTOs becomes needed to have many methods too or many executions flow.

Reply:
Hmm, I don't think so.

The DTO's responsability is to carry data between process; they don't know for what they serve other than hold data.

I used a technique called *named constructor* (or _static factory_) pattern where you can construct one classe in many different ways, using clean and reasonable names instead of only `new`.

Another option would be to construct the `Company` directly in the `CompanyService::register`, but I think that it's much more readable this way.

Here are some resources about that:

http://verraes.net/2014/06/named-constructors-in-php/
https://www.youtube.com/watch?v=DCJJ7rf087Y

Give 'em a shot and let me know your thoughts.

Reply:
Hmm, I don't think so.

The DTO's responsability is to carry data between process; they don't know for what they serve other than hold data.

I used a technique called *named constructor* (or _static factory_) pattern where you can construct one classe in many different ways, using clean and reasonable names instead of only `new`.

Another option would be to construct the `Company` directly in the `CompanyService::register`, but I think that it's much more readable this way.

Here are some resources about that:

http://verraes.net/2014/06/named-constructors-in-php/
https://www.youtube.com/watch?v=DCJJ7rf087Y

Give 'em a shot and let me know your thoughts.

Reply:
I agree with you about the DTO should carry only data, but as I said, to put this logic on the original class isn't a good practice because you will add a responsibility that not belongs to the entity. You can use a Mapper class to do this job.